cmd/podman/create.go

```diff
@@ -67,15 +67,33 @@ func createCmd(c *cli.Context) error {
 		}
 	}
 
-	if len(c.Args()) < 1 {
+	rootfs := c.String("rootfs")
+	if len(c.Args()) < 1 && rootfs == "" {
 		return errors.Errorf("image name or ID is required")
 	}
 
 	mappings, err := util.ParseIDMapping(c.StringSlice("uidmap"), c.StringSlice("gidmap"), c.String("subuidmap"), c.String("subgidmap"))
 	if err != nil {
 		return err
 	}
+	if mappings.UIDMap == nil && os.Getuid() != 0 {
+		uids := []string{fmt.Sprintf("0:%d:1", os.Getuid())}
+		gids := []string{fmt.Sprintf("0:%d:1", os.Getgid())}
+		c.Set("uidmap", uids[0])
+		c.Set("gidmap", gids[0])
+		mappings, err = util.ParseIDMapping(uids, gids, "", "")
```

Comment:
This is a bit ugly. It would be nice for `c.StringSlice("[ug]idmap")` to be parsed _exactly_ once, and the rest should deal with well-typed values (including the initialization of `mappings` here) instead of strings.

Also, separate the implementation logic / defaulting from the CLI integration, e.g. to make testing possible.

Reply:
I've refactored it so that it is done only in `ParseIDMapping`